skymap_scanner/recos/__init__.py

```diff
@@ -79,6 +81,25 @@ def traysegment(self, tray, name, logger, **kwargs: Any) -> None:
         """Performs the reconstruction."""
         pass
 
+    def get_pointing_info(
+        self, 
+        p_frame: I3Frame
+    ) -> Tuple[float, Union[Tuple[float, float], None]]:
+        """Retrieves the direction and opening angle for a pointed scan, if the reco needs it"""
+        ang_dist = 3.5
+        online_ra_dec = None
+        if self.use_pointing:
+            particle_name_possibilities = ["OnlineL2_SplineMPE", "l2_online_SplineMPE"]
+            for particle_name in particle_name_possibilities:
+                if particle_name in p_frame.keys():
+                    online_dir = p_frame[particle_name].dir
+                    online_ra_dec = astro.dir_to_equa(
+                        online_dir.zenith,
+                        online_dir.azimuth,
+                        p_frame["I3EventHeader"].start_time.mod_julian_day_double
+                    )
+        return ang_dist, online_ra_dec
```

Comment:
For this function at large, is its logic specific to the pointed splinempe reco? (not a rhetorical question :smile:) If it is, then it should be put in the derived class--this would also allow you to remove `self.use_pointing`, since it'd rely on inheritance to get the right value(s) instead. The function in the base class would return the trivial values `(3.5, None)`, as it is today.

Reply:
To reply to both comments, I imagine the pointing could be useful not just for SplineMPE, but even for other recos, even if I am not personally planning to implement it for the other recos. In case, different recos could prefer different values of `ang_dist`. For the pointing direction, `OnlineL2_SplineMPE` is what is usually sent out in the first online alert (the first GCN Notice), and I guess would work ad a good direction for pointing independently of the reconstruction algorithm that is after used. 
To make it brief, I believe the optimal solution would be:
1.  A pointing direction independent of the specific reconstruction algorithm (Now SplineMPE_pointed is the only reco algorithm with pointing but in future they could be more);
2.  A `ang_dist` variable which depends on the specific reconstruction.

Reply:
I implemented `ang_dist` as instance of `RecoInterface`. By default `ang_dist` is 3.5. If in future an algorithm needs a particular `ang_dist` this can be specified in the derived class. 

I renamed `online_ra_dec` to `pointing_ra_dec`, so it is more general and it doesn't implicitly say that the pointing direction is `OnlineL2_SplineMPE`. I kept `pointing_ra_dec` as an instance of `PixelsToReco`. Its value is now determined by the function `set_pointing_ra_dec` defined in `recos/__init__.py` but outside the class `RecoInterface`.

I added a the instance `pointing_dir_names` in `RecoInterface`, which by default is `None`. This instance contains the list of names for the `I3Particle` in the `p_frame` which contain the desired pointing direction. This instance is in `SplineMPE_pointed`, and if in future an algorithm will need a different direction this can be specified in the derived class.

The function `get_pointing_info` is now `set_pointing_dir`. It is defined in `recos/__init__.py` but it is not anymore a method of `RecoInterface`. I haven't moved it in `SplineMPE_pointed` because it can work as a general logic for any reco that will need the pointing. To implement the pointing it would just be necessary to define the name of the direction to use with `pointing_dir_names`.